tests/zero-downtime-upgrades/README.md

```diff
@@ -0,0 +1,85 @@
+# Zero-Downtime Upgrades
+
+This document describes a test plan for testing zero-downtime upgrades of NGF.
+
+*Zero-downtime upgrades* means that during an NGF upgrade clients don't experience any
+interruptions to the traffic they send to applications exposed via NGF.
+
+## Goals
+
+- Ensure that upgrading NFG doesn't lead to any loss of traffic flowing through the data plane.
+- Ensure that after an upgrade, NGF can process changes to resources.
```

Comment:
Some guidance from @vepatel might be useful.

Reply:
> This test definitely feels more like a standard test that we would want to run on a more frequent basis, versus strictly a release/performance test.

👍 

However, because customers will upgrade from some previous release, it is important for the NGF to be of a release version before an upgrade -- so that we catch any hidden incompatibilities/dependencies between the old and the new versions that could lead to downtime during an upgrade.